go/database/mpt/proof_test.go

```diff
@@ -401,6 +402,139 @@ func TestWitnessProof_Extract_EmbeddedNode_In_Proof(t *testing.T) {
+	address := common.Address{1}
+	key := common.Key{2}
+	value := common.Value{3}
+
+	ctxt := newNodeContextWithConfig(t, ctrl, S5LiveConfig)
+
+	nonce := common.Nonce{0x01}
+	balance := common.Balance{0x02}
+	hash := common.Hash{0x03}
+	info := AccountInfo{Nonce: nonce, Balance: balance, CodeHash: hash}
+
+	tests := map[string]struct {
+		test func(*testing.T, WitnessProof, common.Hash, common.Address, common.Key) (bool, error)
+	}{
+		"GetAccountInfo": {test: func(t *testing.T, proof WitnessProof, rootHash common.Hash, address common.Address, key common.Key) (bool, error) {
+			got, exist, err := proof.GetAccountInfo(rootHash, address)
+			if !exist || err != nil {
+				return exist, err
+			}
+			if got, want := got, info; got != want {
+				t.Errorf("unexpected account info: got %v, want %v", got, want)
+			}
+			return exist, err
+		}},
+		"GetNonce": {test: func(t *testing.T, proof WitnessProof, rootHash common.Hash, address common.Address, key common.Key) (bool, error) {
+			got, exist, err := proof.GetNonce(rootHash, address)
+			if !exist || err != nil {
+				return exist, err
+			}
+			if got, want := got, nonce; got != want {
+				t.Errorf("unexpected nonce: got %v, want %v", got, want)
+			}
+			return exist, err
+		}},
+		"GetBalance": {test: func(t *testing.T, proof WitnessProof, rootHash common.Hash, address common.Address, key common.Key) (bool, error) {
+			got, exist, err := proof.GetBalance(rootHash, address)
+			if !exist || err != nil {
+				return exist, err
+			}
+			if got, want := got, balance; got != want {
+				t.Errorf("unexpected balance: got %v, want %v", got, want)
+			}
+			return exist, err
+		}},
+		"GetCodeHash": {test: func(t *testing.T, proof WitnessProof, rootHash common.Hash, address common.Address, key common.Key) (bool, error) {
+			got, exist, err := proof.GetCodeHash(rootHash, address)
+			if !exist || err != nil {
+				return exist, err
+			}
+			if got, want := got, hash; got != want {
+				t.Errorf("unexpected code hash: got %v, want %v", got, want)
+			}
+			return exist, err
+		}},
+		"GetValue": {test: func(t *testing.T, proof WitnessProof, rootHash common.Hash, _ common.Address, key common.Key) (bool, error) {
+			got, exist, err := proof.GetState(rootHash, address, key)
+			if !exist || err != nil {
+				return exist, err
+			}
+			if got, want := got, value; got != want {
+				t.Errorf("unexpected value: got %v, want %v", got, want)
+			}
+			return exist, err
+		}},
+	}
+
+	desc := &Extension{
+		path: AddressToNibblePath(address, ctxt)[0:30],
+		next: &Account{address: address, pathLength: 34, info: info,
+			storage: &Extension{
+				path: KeyToNibblePath(key, ctxt)[0:40],
+				next: &Value{key: key, length: 24, value: value},
+			}},
+	}
+
+	root, node := ctxt.Build(desc)
+	totalProof := createReferenceProof(t, ctxt, &root, node)
+
+	handle := node.GetReadHandle()
+	defer handle.Release()
+
+	rootHash, dirty := handle.Get().GetHash()
+	if dirty {
+		t.Fatalf("expected node to be clean")
+	}
```

Comment:
blocks like this could be shortened to
```suggestion
	rootHash, _ := ctxt.getHashFor(&root)
```